test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -1350,21 +1368,46 @@ def get_cgroup_job_dir(self, subsys, jobid, host):
         Returns path of subsystem for jobid
         """
         basedir = self.paths[subsys]
-        # one of these should exist depending on platform
-        # That last option should never exist, but let us
-        # leave it in there
-        jobdirs = [os.path.join(basedir, 'pbs', jobid),
-                   os.path.join(basedir, 'pbs.service/jobid', jobid),
-                   os.path.join(basedir, 'pbs_jobs.service/jobid', jobid),
-                   os.path.join(basedir, 'pbs.slice',
-                                'pbs-%s.slice' % systemd_escape(jobid)),
-                   os.path.join(basedir, 'pbs',
-                                'pbs-%s.slice' % systemd_escape(jobid))]
+        # One of these should exist depending on platform.
+        #
+        # Entries for older hooks deleted. I am leaving them as comments should
+        # this be used to test old hooks, but DO NOT put them at the top
+        # of the list when testing newer hooks, or you will run into
+        # race conditions with some of the hardcoded test timings
+        #
+        # Add these if needed:
+        # os.path.join(basedir, 'pbs.slice',
+        #             'pbs-%s.slice' % systemd_escape(jobid)),
+        # os.path.join(basedir, 'pbs_jobs.slice',
+        #             'pbs_jobs-%s.slice' % systemd_escape(jobid)),
+        # os.path.join(basedir, 'pbs', jobid),
+        # os.path.join(basedir, 'pbs_jobs', jobid),
+        # os.path.join(basedir, 'pbs.service/jobid', jobid),
+        #
+        #
+        # This cleaned version assumes cgroup_prefix is always pbs_jobs,
+        # i.e. cgroup_prefix is not changed
+        #
+        # Separate test for different "sbp" prefix should use
+        # its own script instead; that will not support multi-host jobs
+        jobdirs = [os.path.join(basedir, 'pbs_jobs.service/jobid', jobid)]
         for jdir in jobdirs:
-            if self.du.isdir(hostname=host, path=jdir):
+            if self.du.isdir(hostname=host, path=jdir, sudo=True):
                 return jdir
         return None
 
+    def find_main_cpath(self, cdir):
+        cpath = None
+        if os.path.isdir(cdir):
+            cpath = os.path.join(cdir, 'pbs_jobs.service/jobid')
+            if not os.path.isdir(cpath):
+                cpath = os.path.join(cdir, 'pbs.service/jobid')
+            if not os.path.isdir(cpath):
+                cpath = os.path.join(cdir, 'pbs.slice')
+            if not os.path.isdir(cpath):
+                cpath = os.path.join(cdir, 'pbs')
+        return cpath
```

Comment:
I suggest using iteration to make this function easier to understand
```
def find_main_cpath(self, cdir):
    if os.path.isdir(cdir):
        paths = ['pbs_jobs.service/jobid', 'pbs.service/jobid', 'pbs.slice', 'pbs']
        for p in paths:
            cpath = os.path.join(cdir, p)
            if os.path.isdir(cpath):
                return cpath
    return None
```

Reply:
Quite -- thanks, much better.